src/main/java/com/app/onlinebookstore/service/impl/BookServiceImpl.java

```diff
@@ -38,4 +42,28 @@ public BookDto getById(Long id) {
                 .map(bookMapper::toDto)
                 .orElseThrow(() -> new EntityNotFoundException("Can't find book with id: " + id));
     }
+
+    @Override
+    public BookDto updateById(Long id, CreateBookRequestDto bookRequestDto) {
+        if (bookRepository.findById(id).isEmpty()) {
```

Comment:
Yep, something like that, but more common approach is to fetch an entity and set new fields for this entity and then save it (or not save at all if you want to use hibernate's dirty checking feature, but in your case you are returning the updated entity, so it is better to explicitly save it to return actual data)